src/runtime/ndarray.cc

```diff
@@ -263,6 +263,44 @@ GEN_FROMVECTOR_FOR(uint64_t, kDLInt, 64);
 GEN_FROMVECTOR_FOR(float, kDLFloat, 32);
 GEN_FROMVECTOR_FOR(double, kDLFloat, 64);
 
+template<typename T>
+std::vector<T> NDArray::ToVector(DLDataType dtype) const {
+  CHECK(data_->dl_tensor.ndim == 1) << "ToVector() only supported for 1D arrays";
+  CHECK(data_->dl_tensor.dtype.code == dtype.code &&
+        data_->dl_tensor.dtype.bits == dtype.bits &&
+        data_->dl_tensor.dtype.lanes == dtype.lanes) << "dtype mismatch";
```

Comment:
Do we have an `==` operator overload?

Reply:
https://github.com/dmlc/dgl/blob/master/src/c_api_common.h#L20

Reply:
I moved both overloads to `dgl/runtime/ndarray.h`